packages/ember-glimmer/lib/component.js

```diff
@@ -104,9 +199,282 @@ const Component = CoreView.extend(
       return getAttrFor(attrs, key);
     },
 
+    /**
+      Normally, Ember's component model is "write-only". The component takes a
+      bunch of attributes that it got passed in, and uses them to render its
+      template.
+
+      One nice thing about this model is that if you try to set a value to the
+      same thing as last time, Ember (through HTMLBars) will avoid doing any
+      work on the DOM.
+
+      This is not just a performance optimization. If an attribute has not
+      changed, it is important not to clobber the element's "hidden state".
+      For example, if you set an input's `value` to the same value as before,
+      it will clobber selection state and cursor position. In other words,
+      setting an attribute is not **always** idempotent.
+
+      This method provides a way to read an element's attribute and also
+      update the last value Ember knows about at the same time. This makes
+      setting an attribute idempotent.
+
+      In particular, what this means is that if you get an `<input>` element's
+      `value` attribute and then re-render the template with the same value,
+      it will avoid clobbering the cursor and selection position.
+      Since most attribute sets are idempotent in the browser, you typically
+      can get away with reading attributes using jQuery, but the most reliable
+      way to do so is through this method.
+      @method readDOMAttr
+
+      @param {String} name the name of the attribute
+      @return String
+      @public
+     */
     readDOMAttr(name) {
       return readDOMAttr(this.element, name);
     }
+
+    /**
+      Returns true when the component was invoked with a block template.
+
+      Example (`hasBlock` will be `false`):
+
+      ```hbs
+      {{! templates/application.hbs }}
+      {{foo-bar}}
+      {{! templates/components/foo-bar.hbs }}
+      {{#if hasBlock}}
+        This will not be printed, because no block was provided
+      {{/if}}
+      ```
+
+      Example (`hasBlock` will be `true`):
+
+      ```hbs
+      {{! templates/application.hbs }}
+      {{#foo-bar}}
+        Hi!
+      {{/foo-bar}}
+      {{! templates/components/foo-bar.hbs }}
+      {{#if hasBlock}}
+        This will be printed because a block was provided
+        {{yield}}
+      {{/if}}
+      ```
+
+      This helper accepts an argument with the name of the block we want to check the presence of.
+      This is useful for checking for the presence of the optional inverse block in components.
+
+      ```hbs
+      {{! templates/application.hbs }}
+      {{#foo-bar}}
+        Hi!
+      {{else}}
+        What's up?
+      {{/foo-bar}}
+      {{! templates/components/foo-bar.hbs }}
+      {{yield}}
+      {{#if (hasBlock "inverse")}}
+        {{yield to="inverse"}}
+      {{else}}
+        How are you?
+      {{/if}}
+
+      ```
+      @public
+      @property hasBlock
```

Comment:
We should kill this from the `Ember.Component` documentation, it is very confusing (as folks have reported before).  We need to document `hasBlock` / `has-block` like it was a template helper (like `{{each`, `{{with`, etc).